src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/AvatarsInS3.kt

```diff
@@ -0,0 +1,59 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.jira
+
+import com.atlassian.performance.tools.aws.api.StorageLocation
+import com.atlassian.performance.tools.awsinfrastructure.api.aws.AwsCli
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraHomeSource
+import com.atlassian.performance.tools.ssh.api.SshConnection
+import java.time.Duration
+import java.util.function.Supplier
+
+class AvatarsInS3(
+    private val delegate: JiraHomeSource,
+    private val s3StorageProvider: Supplier<StorageLocation>,
+    private val filestoreConfigAppender: FilestoreConfigAppender
+) : JiraHomeSource {
+    private val associationTarget = "avatars"
+
+    override fun download(
+        ssh: SshConnection
+    ) = delegate
+        .download(ssh)
+        .also { jiraHomeLocation ->
```

Comment:
How would we ensure that uploadToS3 runs only on the shared home (NFS) and that updateFilestoreConfig runs only on application nodes?

Reply:
Could we not ensure that? I was thinking if `aws s3 sync` wouldn't resolve that problem for us. `sync` executed twice shouldn't duplicate any work.